src/main/java/com/simplewebapp/controller/UserController.java

```diff
@@ -16,20 +17,32 @@
  * Created by Zhangariny on 06/07/2015.
  */
 @Controller
+@Path("/user")
 public class UserController {
```

Comment:
You are missing a lot of logging. While writing applications you don't always have access to the source code. Instead logging is there to help you identify easily if this is a code bug or a user error. On public APIs (like a web application) you should always consider coding defensively as you never know who is going to call your API methods. `checkNotNull()`, `log.debug()` and `log.trace()` in some cases are bread and butter on API. Be aware thought in which layer each check should be done. API input is responsible from Controller - and controller should reply back the appropriate msg. Service layer should do internal validation. Finally when a call to DAO happens as a developer you should have done everything possible to prevent a failure (FK checks, not-null checks and whatever the db constrains are).